oscilloscope/oscilloscope_server.py

```diff
@@ -0,0 +1,227 @@
+"""
+### BEGIN NODE INFO
+[info]
+name = Oscilloscope
+version = 0.1
+description = Talks to oscilloscopes
+
+[startup]
+cmdline = %PYTHON% %FILE%
+timeout = 20
+
+[shutdown]
+message = 987654321
+timeout = 20
+### END NODE INFO
+"""
+
+from __future__ import absolute_import
+from labrad.server import setting
+from labrad.gpib import GPIBManagedServer
+from twisted.internet.defer import inlineCallbacks, returnValue
+
+
+import oscilloscope.tektronix as tektronix
+import oscilloscope.agilent as agilent
+
+
+class OscilloscopeServer(GPIBManagedServer):
+    """Manges communication with oscilloscopes. ALL the oscilloscopes."""
+
+    name = 'Oscilloscope Server'
+
+    deviceWrappers = {
+            'AGILENT TECHNOLOGIES DSO-X 4104A': agilent.AgilentDSOX4104AWrapper}
+
+    # SYSTEM
+
+    @setting(11, returns='')
+    def reset(self, c):
+        """Reset the oscilloscope to factory settings."""
+        dev = self.selectedDevice(c)
+        yield dev.reset()
+
+    @setting(12, returns='')
+    def clear_buffers(self, c):
+        """Clear device status buffers."""
+        dev = self.selectedDevice(c)
+        yield dev.clear_buffers()
+
+    # CHANNEL
+
+    @setting(50, channel='i', state='s', returns='s')
+    def channel_on_off(self, c, channel, state=None):
```

Comment:
I'm trying to maintain some level of backward compatibility with existing code. This method is unfortunately called `channel_on_off`, which to me does not mix well with boolean input. I can change it, of course, but it seems funky. On the other hand, I could say that this is a new server with a new interface and change the name to `channel_on` with boolean input. Thoughts?